grpc/percent_to_alteration_test.go

```diff
@@ -0,0 +1,172 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2021 Datadog, Inc.
+
+package grpc_test
+
+import (
+	. "github.com/DataDog/chaos-controller/grpc"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("get mapping from randomly generated Percent to Alteration based on AlterationToPercentAffected", func() {
+	var (
+		altCfgs             []AlterationConfiguration
+		mapping             map[AlterationConfiguration]PercentAffected
+		percentToAlteration []AlterationConfiguration
+	)
+
+	BeforeEach(func() {
+		altCfgs = []AlterationConfiguration{
+			{
+				ErrorToReturn:    "CANCELED",
+				OverrideToReturn: "",
+			},
+			{
+				ErrorToReturn:    "ALREADY_EXISTS",
+				OverrideToReturn: "",
+			},
+			{
+				ErrorToReturn:    "",
+				OverrideToReturn: "{}",
+			},
+		}
+		mapping = make(map[AlterationConfiguration]PercentAffected)
+	})
+
+	Context("with one alteration", func() {
+		It("should create 15 (out of 100 potential) slots with alteration configurations", func() {
+			mapping[altCfgs[0]] = PercentAffected(15)
+
+			percentToAlteration = GetPercentToAlteration(mapping, logger)
+
+			Expect(len(percentToAlteration)).To(Equal(15))
+
+			altCfg := percentToAlteration[0]
+			Expect(altCfg.ErrorToReturn).To(Equal("CANCELED"))
+			Expect(altCfg.OverrideToReturn).To(Equal(""))
+
+			altCfg = percentToAlteration[14]
+			Expect(altCfg.ErrorToReturn).To(Equal("CANCELED"))
+			Expect(altCfg.OverrideToReturn).To(Equal(""))
+		})
+	})
+	Context("with three alterations that add up to less than 100 percent", func() {
+		It("should create 70 (out of 100 potential) slots with alteration configurations", func() {
+			mapping[altCfgs[0]] = PercentAffected(15)
+			mapping[altCfgs[1]] = PercentAffected(20)
+			mapping[altCfgs[2]] = PercentAffected(35)
+
+			percentToAlteration = GetPercentToAlteration(mapping, logger)
+
+			Expect(len(percentToAlteration)).To(Equal(70))
+
+			position := 0
+			altCfg := percentToAlteration[position]
+
+			for i := 0; i < 3; i++ {
```

Comment:
It _may_ be simpler to loop through every item in `percentToAlteration` and just count how many times each error is set? Then compare to the original percents specified?